src/pages/workspace/perDiem/WorkspacePerDiemPage.tsx

```diff
@@ -196,6 +195,26 @@ function WorkspacePerDiemPage({route}: WorkspacePerDiemPageProps) {
         [allSubRates, selectedPerDiem, canSelectMultiple, styles.flex2, styles.alignItemsStart, styles.textSupporting, styles.label, styles.pl2, styles.alignSelfEnd],
     );
 
+    const [, startTransition] = useTransition();
+    const [filteredSubRatesList, setFilteredSubRatesList] = useState<PolicyOption[]>([]);
+
+    useEffect(() => {
+        startTransition(() => {
+            const normalizedSearchQuery = inputValue.trim().toLowerCase();
+            const filtered = normalizedSearchQuery ? subRatesList.filter((subRate) => subRate.text?.toLowerCase()?.includes(normalizedSearchQuery)) : subRatesList;
+            setFilteredSubRatesList(lodashSortBy(filtered, 'text', localeCompare) as PolicyOption[]);
+        });
+    }, [inputValue, subRatesList]);
+
+    // Filter out rates that will be deleted
+    const allSelectableSubRates = useMemo(() => {
```

Comment:
NAB move this into state and use `useTransition` for it as well, since it's affected by changes to `inputValue`